third-party/qthread/qthread-src/src/feb.c

```diff
@@ -175,7 +175,11 @@ static inline void qt_feb_schedule(qthread_t          *waiter,
 {
     qthread_debug(FEB_DETAILS, "waiter(%p:%i), shep(%p:%i): setting waiter to 'RUNNING'\n", waiter, (int)waiter->thread_id, shep, (int)shep->shepherd_id);
     waiter->thread_state = QTHREAD_STATE_RUNNING;
-    if ((waiter->flags & QTHREAD_UNSTEALABLE) && (waiter->rdata->shepherd_ptr != shep)) {
+    // Chapel hack -- we either use schedulers that don't support work stealing
+    // (nemesis), or we run with stealing disabled (distrib w/ STEAL_RATIO=0).
+    // Spawning to the current shepherd below serializes execution and relies
+    // on work stealing to redistribute, so just always spawn to the orig shep
+    if (1) { //(waiter->flags & QTHREAD_UNSTEALABLE) && (waiter->rdata->shepherd_ptr != shep)) {
```

Comment:
Rather a nit, but this seems more clear to me:

    if (1 || ((waiter->flags & QTHREAD_UNSTEALABLE) && (waiter->rdata->shepherd_ptr != shep))) {

The target compiler should be able to constant-fold this at compile time, so no perf implications. It's okay if you don't want to change it, though.